### PR TITLE
test: add a babel example

### LINKED ISSUE #539: end-to-end testing w/ babel
using async functions. we've tested generators but not async functions

### LINKED ISSUE #572: Documentation should include an example of babel configuration
Now with Babel 6, things got a little more complex:

```
// index.js
require("babel-polyfill");
require('babel-core/register')({
    presets: ["es2015"],
    plugins: ["syntax-async-functions", "syntax-async-generators"]
});

require('./app.js');
```

### PR DESCRIPTION
**Pull request by jonathanong**
closes #539
closes #572

TODO:

- [x] docs linking to `require('babel-core/register')` + `stage-3` preset or babel
- [x] figuring out why `async (ctx, next) => {}` doesn't work. anyone know why it doesn't work?

### CODE REVIEW

**Comment by jonathanong** [comment]
it's fixed now... waiting on https://github.com/babel/babel-eslint/issues/199 for linting

File: test/babel/.babelrc [hunk 1]
```diff
@@ -0,0 +1,3 @@
+{
+  "presets": ["stage-3"]
```
Comment (broucz) [inline test/babel/.babelrc]: I wonder if using `stage-x` is ideal. I mean, someone know if `stage` presets will evolve in time with ESxx ?
If it's the case, using transformers directly could be better no ?

```js
{
  "plugins": ["transform-async-to-generator"]
}
```

File: test/babel/index.js [hunk 2]
```diff
@@ -0,0 +1,6 @@
+'ues strict';
```
Comment (broucz) [inline test/babel/index.js]: typo here

File: test/babel/.babelrc (hunk 1 above)
Comment (menems) [inline test/babel/.babelrc]: +1 stage-X is too global

Comment (jonathanong) [inline test/babel/.babelrc]: SGTM!

**Comment by jonathanong** [comment]
for review, guys!

**Comment by fengmk2** [comment]
LGTM

**Pull request merged**